library/ssl_tls13_generic.c

```diff
@@ -1308,123 +1304,37 @@ int mbedtls_increment_sequence_number( unsigned char *sequenceNumber, unsigned c
-     *
-     * The hash of the structure is returned in the variable output_hash.
-     */
-static int ssl_calc_verify_tls_sha256( mbedtls_ssl_context *ssl, unsigned char output_hash[32], int from )
-{
-    int ret = 0;
-
-    /* The length of context_string_[client|server] is
-     * sizeof( "TLS 1.3, xxxxxx CertificateVerify" ) - 1, i.e. 33 bytes.
-     */
-    const unsigned int content_string_len = sizeof( "TLS 1.3, xxxxxx CertificateVerify" ) - 1;
-    const unsigned char context_string_client[] = "TLS 1.3, client CertificateVerify";
-    const unsigned char context_string_server[] = "TLS 1.3, server CertificateVerify";
-    mbedtls_sha256_context sha256;
-    unsigned char handshake_hash[ 32 ];
-    unsigned char verify_buffer[ 64 + 33 + 1 + 32 ];
-
-    mbedtls_sha256_init( &sha256 );
-
-    MBEDTLS_SSL_DEBUG_MSG( 2, ( "=> calc verify sha256" ) );
-
-    mbedtls_sha256_clone( &sha256, &ssl->handshake->fin_sha256 );
-
-    if( ( ret = mbedtls_sha256_finish_ret( &sha256, handshake_hash ) ) != 0 )
-    {
-        MBEDTLS_SSL_DEBUG_RET( 1, "mbedtls_sha256_finish_ret", ret );
-        goto exit;
-    }
-
-    MBEDTLS_SSL_DEBUG_BUF( 3, "handshake hash", handshake_hash, 32 );
-
-    memset( verify_buffer, 32, 64 );
-
-    if( from == MBEDTLS_SSL_IS_CLIENT )
-    {
-        memcpy( verify_buffer + 64, context_string_client, content_string_len );
-    }
-    else /* from == MBEDTLS_SSL_IS_SERVER */
-    {
-        memcpy( verify_buffer + 64, context_string_server, content_string_len );
-    }
-
-    verify_buffer[64 + content_string_len] = 0x0;
-    memcpy( verify_buffer + 64 + content_string_len + 1, handshake_hash, 32 );
-
-    MBEDTLS_SSL_DEBUG_BUF( 3, "verify buffer", verify_buffer, 64 + content_string_len + 1 + 32 );
-
-    mbedtls_sha256( verify_buffer, 64 + content_string_len + 1 + 32, output_hash, 0 /* 0 for SHA-256 instead of SHA-224 */ );
-
-    MBEDTLS_SSL_DEBUG_BUF( 3, "verify hash", output_hash, 32 );
-
-    MBEDTLS_SSL_DEBUG_MSG( 2, ( "<= calc verify" ) );
-
-exit:
-    mbedtls_sha256_free( &sha256 );
-    return( ret );
-}
-#endif /* MBEDTLS_SHA256_C */
-
-#if defined(MBEDTLS_SHA512_C)
-    /*
-     * ssl_calc_verify_tls_sha384() computes hash over a structure,
-     * which is later digitally signed and placed into the CertificateVerify message.
+     *   - 32 or 48 bytes for the Transcript-Hash(Handshake Context, Certificate)
+     *     (depending on the size of the transcript_hash)
      *
-     * The structure computed in this function is:
-     *   - 64 bytes of octet 32,
-     *   - 33 bytes for the context string (which is either "TLS 1.3, client CertificateVerify"
-     *     or "TLS 1.3, server CertificateVerify"),
-     *   - 1 byte for the octet 0x0, which servers as a separator,
-     *   - 48 bytes for the Transcript-Hash(Handshake Context, Certificate)
+     * The size of the verify_structure is assumed to be either
+     * - 130 bytes for a SHA256-based transcript hash, or
```

Comment:
Reading this for the first time, I wonder where those numbers come from. Perhaps the documentation could be expanded a bit here?

Reply:
I modified the wording to make this more clear.